sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -955,6 +957,18 @@ class Analyzer(
         Project(p.output, newPlan.withNewChildren(newChild :: Nil))
     }
   }
+
+  object CleanAlias extends Rule[LogicalPlan] {
+    override def apply(plan: LogicalPlan): LogicalPlan = plan transformAllExpressions  {
+      case UnresolvedAlias(child) => child match {
+        case g: GetStructField => Alias(g, g.field.name)()
```

Comment:
Sorry I made a mistake here, we should just remove the `UnresolvedAlias`, just like what we did in `Analyzer.ResolveReferences`(we call `trimUnresolvedAlias` there).

The key problem is we have 2 code path to resolve `UnresolvedAttribute`, one is `Analyzer.ResolveReferences`(last case) and the other is `DataFrame.resolve`, so we need to make them consistent, i.e. call `trimUnresolvedAlias` at `DataFrame.resolve` or abstract this logic for sql and dataframe.